config/v1/types_cluster_version.go

```diff
@@ -756,3 +786,26 @@ type ClusterVersionList struct {
 
 	Items []ClusterVersion `json:"items"`
 }
+
+// SignatureStore represents the URL of custom Signature Store
+type SignatureStore struct {
+
+	// url contains the upstream custom signature store URL.
+	// url should be a valid absolute http/https URI of an upstream signature store as per rfc1738.
+	// This must be provided and cannot be empty.
+	//
+	// +kubebuilder:validation:Type=string
+	// +kubebuilder:validation:XValidation:rule="isURL(self)",message="url must be a valid absolute URL"
+	// +kubebuilder:validation:Required
+	URL string `json:"url"`
+
+	// ca is an optional reference to a config map by name containing the PEM-encoded CA bundle.
+	// It is used as a trust anchor to validate the TLS certificate presented by the remote server.
+	// The key "ca.crt" is used to locate the data.
+	// If specified and the config map or expected key is not found, the signature store is not honored.
+	// If the specified ca data is not valid, the signature store is not honored.
+	// If empty, we fall back to the CA configured via Proxy, which is appended to the default system roots.
+	// The namespace for this config map is openshift-config.
+	// +optional
+	CA ConfigMapNameReference `json:"ca"`
```

Comment:
Should this be `omitempty` given it is optional?

Reply:
I don't think `omitempty` does anything for non-pointer Go structs like this: golang/go#11939.  I'm fine with this property becoming an `omitempty` pointer, but none of the existing consumers are using pointers.

Reply:
Thanks, makes sense.

Reply:
@PratikMahajan Can you run a quick test for me, what is the marshalled version of `SignatureStore{}`, and does the API accept that? I think it will be
```
- url: ""
  ca:
    name: ""
```
Which will pass the required checks., but probably fail on the `isURL` check. Assuming I'm correct, the CA can stay as it is in terms of marshalling

Reply:
There are many similar examples in the existing code which points to similar patterns e.g. 

https://github.com/openshift/api/blob/master/config/v1/types_oauth.go#L263
https://github.com/openshift/api/blob/master/config/v1/types_oauth.go#L328
https://github.com/openshift/api/blob/master/config/v1/types_oauth.go#L456
https://github.com/openshift/api/blob/master/config/v1/types_oauth.go#L481
https://github.com/openshift/api/blob/master/config/v1/types_oauth.go#L519